### PR TITLE
Dev 8

### PR DESCRIPTION
**Pull request by ohatsenko**

### CODE REVIEW

**Review by ivan0dyatlyukkk — APPROVED** [review]
GJ!

File: src/main/java/com/example/onlinebookstore/model/CartItem.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@SQLDelete(sql = "UPDATE cart_item SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "cart_item")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "shopping_cart_id", nullable = false)
+    private ShoppingCart shoppingCart;
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/model/CartItem.java:30]: Please make toOne relations lazy

File: src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Override
+    public void delete(Long id) {
+        cartItemRepository.deleteById(id);
+    }
+
+    @Override
+    public CartItemDto save(CreateCartItemRequestDto requestDto, Long id) {
+        CartItem cartItem = new CartItem();
+        cartItem.setBook(bookRepository.findById(requestDto.getBookId()).orElseThrow(
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java]: Please use book service to get entities via repository

File: src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    @Override
+    public Set<CartItemDto> findByShoppingCart(Long id) {
+        return cartItemRepository.findCartItemsByShoppingCartId(id).stream()
+                .map(cartItemMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Long id) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(id)
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java]: code duplication

File: src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        return cartItemMapper.toUpdateDto(cartItem);
+    }
+
+    @Override
+    public void delete(Long id) {
+        cartItemRepository.deleteById(id);
+    }
+
+    @Override
+    public CartItemDto save(CreateCartItemRequestDto requestDto, Long id) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java]: parameter id is super confusing, please rename it

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -11,3 +11,13 @@ databaseChangeLog:
...
+    - include:
+          file: db/changelog/changes/007-create-category-table.yaml
+    - include:
+          file: db/changelog/changes/008-create-book-category-table.yaml
+    - include:
+          file: db/changelog/changes/009-create-shopping-cart-table.yaml
+    - include:
+        file: db/changelog/changes/010-create-cart-item-table.yaml
+    - include:
+        file: db/changelog/changes/011-create-shopping-cart-cart-item-table.yaml
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: add empty line

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ, I left minor comments

**Review by andrii-hoienko — APPROVED** [review]
GJ!

**Pull request merged**